apps/web/src/components/borrow/modal/DepositModal.tsx

```diff
@@ -0,0 +1,263 @@
+  const signer = provider.getSigner()
+
+  const [hasError, setHasError] = useState()
+  const [loading, setLoading] = useState(false)
+
+  const [shouldApproveContract, setShouldApproveContract] = useState(false)
+  const [approved, setApproved] = useState<boolean>(false)
+  const [healthFactor, setHealthFactor] = useState<number>()
+
+  const [depositCompleted, setDepositCompleted] = useState<boolean>(false)
+  const tokenContract = reserveData.symbol
+    ? Tokens.getTokenContract(signer, reserveData.symbol)
+    : null
+  const lendingPoolContract =
+    BorrowAndLending.lendingPool.getLendingPoolContract(signer)
+
+  const isApproveDisabled =
+    !isSupportedNetwork || loading || approved || !reserveData.balance
+
+  const checkAllowance: CheckAllowanceFunction = useCallback(async () => {
+    if (tokenContract) {
+      try {
+        const approvedTokens = await Tokens.getAllowance(
+          tokenContract,
+          currentAccount,
+          'deposit',
+        )
+        setShouldApproveContract(approvedTokens.toString() === '0')
+        setHasError(null)
+      } catch (error) {
+        setHasError(error)
+      }
+    }
+  }, [currentAccount, tokenContract])
+
+  useEffect(() => {
+    if (isSupportedNetwork) {
+      void checkAllowance()
+    }
+  }, [checkAllowance, isSupportedNetwork])
+
+  const getHealthFactor = async () => {
+    setLoading(true)
+    try {
+      const response = await BorrowAndLending.lendingPool.getUserAccountData(
+        lendingPoolContract,
+        currentAccount,
+      )
+      setHealthFactor(Number(ethers.utils.formatUnits(response.healthFactor)))
+    } catch (error) {
+      setHasError(error)
+    } finally {
+      setLoading(false)
+    }
+  }
+  useEffect(() => {
+    void getHealthFactor()
+  }, [])
+
+  const approve = async () => {
+    setLoading(true)
+    try {
+      const response = await Tokens.approveToken(tokenContract, 'deposit')
+      const isApproved = await response.wait()
+
+      setApproved(!!isApproved)
+    } catch (error) {
+      setHasError(error)
+    } finally {
+      setLoading(false)
+    }
+  }
+
+  const deposit = async (amount: number) => {
+    setLoading(true)
+    try {
+      const response = await BorrowAndLending.lendingPool.deposit(
+        lendingPoolContract,
+        reserveData.underlyingAsset,
+        amount,
+        currentAccount,
+      )
+      const isDeposited = await response.wait()
+      setDepositCompleted(!!isDeposited)
+    } catch (error) {
+      setHasError(error)
+    } finally {
+      setLoading(false)
+    }
+  }
+
+  const depositValidationSchemaa = Yup.object().shape({
+    amount: Yup.number()
+      .typeError(t('global.errors.numbersOnly'))
+      .positive(t('global.errors.positiveValue'))
+      .max(Number(reserveData.balance), t('global.errors.exceededBalance'))
+      .required(t('global.errors.required')),
+  })
+
+  return (
+    <Modal isOpen={isOpen} onClose={onClose}>
+      <Formik
```

Comment:
we should use `useFormik` <Formik as a container is not good regarding performance.

Reply:
Something like this on slaff.
`apps/iam/src/components/auth/UserInfoStep.tsx`